test/integration/builder/selects.js

```diff
@@ -540,6 +540,45 @@ module.exports = function(knex) {
           });
       });
 
+      it.only('allows parameter to be a query builder', function() {
```

Comment:
`.only` should be removed and a bit more complex test cases with multiple conditions and nested subqueries should be added. 

Something like 

```
simpleWhereClause = knex.where('id', 1);

complexWhereClause  = knex
  .where('active', true)
  .whereIn('id', function () { 
    this.from('other').whereExist('imNotNull').where(simpleWhereBuilder).where(function () {
      this.where(simpleWhereBuilder);
    });
  });

knex('accounts').where(simpleWhereClause).where(complexWhereClause).orWhere(simpleWhereClause);
```